src/test/java/org/crumbleworks/forge/crumbutil/configurations/CrumbHomeTest.java

```diff
@@ -1,146 +1,139 @@
-    
+
     private static String DEFAULT_HOME_TEST_PATH = TEST_DIR + File.separator + ".crumbleworks"; //where the CrumbHome should point to per default settings
-    
+
     private static String CRUMBLEWORKS_HOME_ENV_VAR = "CRUMBLEWORKS_HOME";
     private static String CRUMBLEWORKS_HOME_TEST_PATH = TEST_DIR + File.separator + "keks_Haus"; //an alternate CrumbHome location provided using the envVar configuration
 
-    //[A-Za-z0-9-_]+ (see CrumbHome.CRUMBLEWORKS_APP_NAME_REGEX)
-    private static String[] validAppNames = {"Warrior2D", "Crumb2Crumb", "CrumbUtil", "CrumbEngine", "CrumbGame-Alpha", "graPHity", "Hindenburg", "Dont_Die"};
-    private static String[] invalidAppNames = {"Peter, bist du da?", "Als Joghurt fliegen lernte", "?!?", ". I'm not fat, just <bold>!"};
-    
-    @Before
-    public void everytimeSetup() throws IOException {
-        Files.createDirectories(Paths.get(USER_HOME_TEST_PATH));
-        
-        mockStatic(System.class);
-        
-        //return a 'test' path as 'user.home' so the underlying system is not touched
-        when(System.getProperty(USER_HOME_SYSTEM_PROPERTY_KEY)).thenReturn(USER_HOME_TEST_PATH);
-    }
-
-    @After
+    @AfterEach
     public void everytimeCleanup() throws IOException {
         FileUtil.deepRemoveFolder(Paths.get(USER_HOME_TEST_PATH));
     }
-    
-    @Test
-    public void getAppHomeWithoutEnvVarSetWithValidAppNames() throws IOException {
-        //return 'null' when requesting the environment variable, simulating it not being there
-        when(System.getenv(CRUMBLEWORKS_HOME_ENV_VAR)).thenReturn(null);
 
-        AppHome appHome = null;
-        
-        for(String appName : validAppNames) {
-            //check creating the appHome from clean-slate
-            appHome = CrumbHome.get(appName);
-            assertEquals(appName, appHome.getApp());
-            assertEquals(Paths.get(DEFAULT_HOME_TEST_PATH + File.separator + appName), appHome.getPath());
-            assertEquals(appName + "@" + Paths.get(DEFAULT_HOME_TEST_PATH) + File.separator + appName, appHome.toString());
-            
-            //repeat, this time retrieving the just created appHome
-            appHome = CrumbHome.get(appName);
-            assertEquals(appName, appHome.getApp());
-            assertEquals(Paths.get(DEFAULT_HOME_TEST_PATH + File.separator + appName), appHome.getPath());
-            assertEquals(appName + "@" + Paths.get(DEFAULT_HOME_TEST_PATH) + File.separator + appName, appHome.toString());
-        }
+    @ParameterizedTest
+    @ValueSource(strings = {"Warrior2D", "Crumb2Crumb", "CrumbUtil", "CrumbEngine", "CrumbGame-Alpha", "graPHity", "Hindenburg", "Dont_Die"})
+    public void getAppHomeWithoutEnvVarSetWithValidAppNames(String appName) throws Exception {
+        restoreSystemProperties(() -> {
+            System.setProperty(USER_HOME_SYSTEM_PROPERTY_KEY, USER_HOME_TEST_PATH);
```

Comment:
die zile chenntisch easily extrahiere in es @foreach oder wie's bi junit5 heisst. s bruchts ja defo i jedem test. somit reduzieremer de testcontent uf das wo eefektiv testet wird / unterschidlech isch

Reply:
McDonnough sagt es geht nicht